yt/utilities/lib/geometry_utils.pyx

```diff
@@ -517,48 +517,60 @@ def triangle_plane_intersect(int ax, np.float64_t coord,
     for i in range(ntri):
         count = 0
 
-        # skip if triangle is close to being parallel to plane
+        # Here p0 holds the triangle's zeroth node coordinates,
+        # p1 holds the first node's coordinates, and
+        # p2 holds the second node's coordinates
         for j in range(3):
             p0[j] = triangles[i, 0, j]
             p1[j] = triangles[i, 1, j]
-            p3[j] = triangles[i, 2, j]
+            p2[j] = triangles[i, 2, j]
             plane_norm[j] = 0.0
         plane_norm[ax] = 1.0
         subtract(p1, p0, E0)
-        subtract(p3, p0, E1)
+        subtract(p2, p0, E1)
         cross(E0, E1, tri_norm)
         dp = dot(tri_norm, plane_norm)
         dp /= L2_norm(tri_norm)
+        # Skip if triangle is close to being parallel to plane.
         if (fabs(dp) > 0.995):
             continue
-        
+
         # Now for each line segment (01, 12, 20) we check to see how many cross
-        # the coordinate.
+        # the coordinate of the slice.
+        # Here, the components of p2 are either +1 or -1 depending on whether the
+        # node's coordinate corresponding to the slice axis is greater than the
+        # coordinate of the slice. p2[0] -> node 0; p2[1] -> node 1; p2[2] -> node2
         for j in range(3):
-            p3[j] = copysign(1.0, triangles[i, j, ax] - coord)
-        if p3[0] * p3[1] < 0: count += 1
-        if p3[1] * p3[2] < 0: count += 1
-        if p3[2] * p3[0] < 0: count += 1
+            p2[j] = copysign(1.0, triangles[i, j, ax] - coord + 0)
+        if p2[0] * p2[1] < 0: count += 1
+        if p2[1] * p2[2] < 0: count += 1
+        if p2[2] * p2[0] < 0: count += 1
         if count == 2:
             nlines += 1
         elif count == 3:
-            nlines += 2
+            raise RuntimeError("I'm impressed you got a plane to intersect all three "
+                               "legs of a triangle")
```

Comment:
Can you reword this? Imagine you're a random user who hits this error and doesn't understand the guts of what's going on here.